draft-irtf-panrg-path-properties.md

```diff
@@ -196,9 +200,13 @@ Service function:
 : A service function that a path element applies to a flow, see {{RFC7665}}. Examples of abstract service functions include firewalls, Network Address Translation (NAT), and TCP optimizers. Some stateful service functions, such as NAT, need to observe the same flow in both directions, e.g., by being an element of both the path and the reverse path.
 
 Transparency:
-: A node is transparent with respect to a protocol header, payload, or both for a specific action if the node performs this action independently of the given (meta-)information.
+: A node is transparent to a flow with respect to some (meta-)information for an action if the node performs this action independently of the given (meta-)information.
+(Meta-)information can for example be the existence of a protocol (header) in a packet or the content of a protocol header, payload, or both.
 Actions can for example be blocking packets or reading and modifying (other protocol) headers or payloads.
-An IP router could be transparent for transport protocol headers such as TCP and UDP, in contrast to a NAT that actively modifies TCP and UDP header information.
+An IP router could be transparent to transport protocol headers such as TCP/UDP but not transparent to IP headers as it actively modifies them.
+A TCP SYN flooding firewall that drops TCP SYN packets after reaching a limit is transparent to IP but not to TCP/UDP.
```

Comment:
If the firewall indeed does not look at the IP addresses at all, it would help to clarify the example, e.g.:
```suggestion
A TCP SYN flooding firewall that drops TCP SYN packets after having seen a certain number of packets with TCP SYN, regardless of IP address, is transparent to IP but not to TCP/UDP.
```

But it doesn't sound like a good idea to implement it this way, as that means you could DoS a network by SYN-flooding it. 
So perhaps this firewall should count TCP SYN packets from a particular IP address or for a combination of source and destination IP addresses, in which case it would not be transparent to either TCP or IP.

What do you think?

Reply:
I wanted to have an example where a node is transparent to IP but not to TCP/UDP to show that such a combination is also possible.
But I agree that the example does not make much sense in practice ^^

Maybe we could use the example of a firewall that only allows outgoing TCP connection, i.e., blocks all incoming TCP SYN packets regardless of the IP address?
That seems more reasonable.

Reply:
Yep, sounds like this is a better example.